var/spack/repos/builtin/packages/raja/package.py

```diff
@@ -246,19 +308,50 @@ def initconfig_package_entries(self):
         else:
             entries.append(cmake_cache_option("ENABLE_EXERCISES", "+exercises" in spec))
 
+        # TODO: Treat the workaround when building tests with spack wrapper
+        #       For now, removing it to test CI, which builds tests outside of wrapper.
         # Work around spack adding -march=ppc64le to SPACK_TARGET_ARGS which
         # is used by the spack compiler wrapper.  This can go away when BLT
         # removes -Werror from GTest flags
-        if self.spec.satisfies("%clang target=ppc64le:") or not self.run_tests:
+        #
+        # if self.spec.satisfies("%clang target=ppc64le:")
+        #   or (not self.run_tests and "+tests" not in spec):
+        if not self.run_tests and "+tests" not in spec:
             entries.append(cmake_cache_option("ENABLE_TESTS", False))
         else:
             entries.append(cmake_cache_option("ENABLE_TESTS", True))
+            if "+run-all-tests" not in spec:
+                if spec.satisfies("%clang@12.0.0:13.9.999"):
+                    entries.append(
+                        cmake_cache_string(
+                            "CTEST_CUSTOM_TESTS_IGNORE",
+                            "test-algorithm-sort-OpenMP.exe;test-algorithm-stable-sort-OpenMP.exe",
+                        )
+                    )
+                excluded_tests = [
+                    "test-algorithm-sort-Cuda.exe",
+                    "test-algorithm-stable-sort-Cuda.exe",
+                    "test-algorithm-sort-OpenMP.exe",
+                    "test-algorithm-stable-sort-OpenMP.exe",
+                ]
+                if spec.satisfies("+cuda %clang@12.0.0:13.9.999"):
+                    entries.append(
+                        cmake_cache_string("CTEST_CUSTOM_TESTS_IGNORE", ";".join(excluded_tests))
+                    )
+                if spec.satisfies("+cuda %xl@16.1.1.12"):
+                    entries.append(
+                        cmake_cache_string(
+                            "CTEST_CUSTOM_TESTS_IGNORE",
+                            "test-algorithm-sort-Cuda.exe;test-algorithm-stable-sort-Cuda.exe",
+                        )
+                    )
```

Comment:
@rhornung67 I’m not sure why this is considered pure new addition. Maybe it was removed from RAJA package in Spack upstream ? I don’t think it was specific to our local version. I need to check the history.

Reply:
This is likely a new addition :)